IBX-6404: Added create bundle extension (#2214)
ibexa/documentation-developer: docs/administration/project_organization/bundle_extension/install_package.md

---

> description: Install created bundle extension into Ibexa DXP.

Reviewer: ```suggestion
description: Install a bundle extension you created into Ibexa DXP.
```

---

# Install bundle
 
## Add repository to composer

> 
> To be able to install the bundle in your [[= product_name_base =]] project, add the repository to the `composer.json`:

Reviewer: ```suggestion
To install the bundle in your [[= product_name_base =]] project, add the repository to `composer.json`:
```

```json hl_lines="17"
    "repositories": {
        "ibexa": {
            "type": "composer",
            "url": "https://updates.ibexa.co"
        },
        "acme/currency-exchange-rate":{
            "type": "vcs",
            "url": "file:///Users/justyna.koralewicz/example-3rd-party-extension"
        }
    }
```

## Install bundle into application

> On your [[= product_name_base =]] project root run:

Reviewer: ```suggestion
In your [[= product_name_base =]] project root directory, run:
```

 
```bash
composer require acme/currency-exchange-rate:dev-main
```

!!! note

>     If your application uses Symfony Flex, the bundle is registered automatically after you install it.
> Check whether the bundle is enabled, if not, you must enable it per environment in the `config/bundles.php` file:

Reviewer: ```suggestion
    If your application uses Symfony Flex, the bundle is registered automatically after you install it.
    Check whether the bundle is enabled.
    If not, you must enable it per environment in the `config/bundles.php` file:
```

```php
return [
    Symfony\Bundle\FrameworkBundle\FrameworkBundle::class => ['all' => true],
    ...
    Ibexa\Bundle\TestFixtures\IbexaTestFixturesBundle::class => ['all' => true],
    ACME\Bundle\CurrencyExchangeRate\ACMECurrencyExchangeRateBundle::class => ['all' => true],
];
```

> Next, clear the cache by runnig the following command:

Reviewer: ```suggestion

Next, clear the cache by running the following command:
```

```bash
php bin/console cache:clear
```

> The newly installed bundle should be visible in the **Composer** tab in **Admin** -> **System information**.

Reviewer: ```suggestion
The newly installed bundle should be visible in **Admin** -> **System information** -> **Composer** tab.
```

![Installed bundles](sys_info_composer_tab.png)

## Add currency exchange page block

> Go to Page Builder edit mode. The Currency exchange block should be visible and available in the **Elements** panel.

Reviewer: ```suggestion
Go to Page Builder edit mode. 
The Currency exchange block should be visible and available in the **Elements** panel.
```

---
description: Install created bundle extension into Ibexa DXP.
---

# Install bundle
 
## Add repository to composer

> To be able to install the bundle to your [[= product_name_base =]] project, add the repository to the `composer.json`:

Reviewer 2: ```suggestion
To be able to install the bundle in your [[= product_name_base =]] project, add the repository to the `composer.json`:
```

```json hl_lines="17"
    "repositories": {
        "ibexa": {
            "type": "composer",
            "url": "https://updates.ibexa.co"
        },
        "acme/currency-exchange-rate":{
            "type": "vcs",
            "url": "file:///Users/justyna.koralewicz/example-3rd-party-extension"
        }
    }
```

## Install bundle into application

On your [[= product_name_base =]] project root run:
 
```bash
composer require acme/currency-exchange-rate:dev-main
```

!!! note

    If your application uses Symfony Flex, the bundle is registered automatically after you install it.

> 
> 

Reviewer 2: ```suggestion

```

Check whether the bundle is enabled, if not, you must enable it per environment in the `config/bundles.php` file:

> 
> 

Reviewer 2: ```suggestion

```

```php
return [
    Symfony\Bundle\FrameworkBundle\FrameworkBundle::class => ['all' => true],
    ...
    Ibexa\Bundle\TestFixtures\IbexaTestFixturesBundle::class => ['all' => true],
    ACME\Bundle\CurrencyExchangeRate\ACMECurrencyExchangeRateBundle::class => ['all' => true],
];
```

> 
> 

Reviewer 2: ```suggestion

```

Next, clear the cache by runnig the following command:

```bash
php bin/console cache:clear
```

The newly installed bundle should be visible in the **Composer** tab in **Admin** -> **System information**.

![Installed bundles](sys_info_composer_tab.png)

> 
> 

Reviewer 2: ```suggestion

```

## Add currency exchange page block

Go to Page Builder edit mode. The Currency exchange block should be visible and available in the **Elements** panel.

> 
> 

Reviewer 2: ```suggestion

```
